pkg/frontend/mysql_cmd_executor.go

```diff
@@ -265,9 +264,15 @@ var RecordStatement = func(ctx context.Context, ses *Session, proc *process.Proc
 	return motrace.ContextWithStatement(trace.ContextWithSpanContext(ctx, sc), stm)
 }
 
-var RecordParseErrorStatement = func(ctx context.Context, ses *Session, proc *process.Process, envBegin time.Time, envStmt, sqlType string) context.Context {
-	for _, sql := range parsers.SplitSqlBySemicolon(envStmt) {
+var RecordParseErrorStatement = func(ctx context.Context, ses *Session, proc *process.Process, envBegin time.Time, envStmt string, sqlTypes []string, err error) context.Context {
+	retErr := moerr.NewParseError(ctx, err.Error())
+	sqlType := sqlTypes[0]
+	for i, sql := range parsers.SplitSqlBySemicolon(envStmt) {
+		if i < len(sqlTypes) {
+			sqlType = sqlTypes[i]
+		}
 		ctx = RecordStatement(ctx, ses, proc, nil, envBegin, sql, sqlType, true)
+		motrace.EndStatement(ctx, retErr, 0)
```

Comment:
normally, it just need to call once in function `logStatementStringStatus`.
but here it generates multi statement_info, so it needs to call `EndStatement` at the same time.